shared/utils/TasksProvider/TasksProvider.tsx

```diff
@@ -30,16 +33,15 @@ interface IProps {
 }
 
 function TasksProvider({ children }: IProps) {
-  const [tasks, setTasks] = useState<TTask[]>([]);
   const [history, setHistory] = useState<TTask[][]>([[]]);
   const [currStateIndex, setCurrStateIndex] = useState(0);
   const [filter, setFilter] = useState<TFilter | null>(null);
-  const { data } = useGetAllTasksQuery();
+  const { data: tasks, refetch } = useGetAllTasksQuery();
+  const [createTask] = useCreateTaskMutation();
```

Comment:
The context should be used for state management, not for api calls. Please create a Tech Debt ticket to move all the api calls and hooks to their relevant components.